meteor/lib/rundown/infinites.ts

```diff
@@ -153,6 +177,33 @@ export function getPlayheadTrackingInfinitesForPart(
 								(segmentsBeforeThisInRundownSet.has(currentPartInstance.segmentId) ||
 									currentPartInstance.segmentId === part.segmentId)
 							break
+						case PieceLifespan.OutOnShowStyleEnd:
+							if (!rundownsBeforeThisInPlaylistSet.has(currentPartInstance.rundownId)) {
+								isValid = false
+								break
+							}
+
+							if (candidatePiece.rundownId === part.rundownId) {
+								isValid = true
+								break
+							}
+
+							const rundownsBeforeThisInPlaylist = Array.from(rundownsBeforeThisInPlaylistSet.values())
```

Comment:
I believe that converting a set into an array will result in an array with an unpredictable order. Also I dont see how we know that `prevRundownId` is related to `currentPartInstance` or the new instance.
From reading this block of code, it looks like we are checking that `currentPartInstance` is somewhere before the new part, and then checking that the rundown before the new part (the two arent guaranteed to be the same) is of the correct showstyle. 

Q) What happens if we have 4 rundowns?
  Rd1 = 'currentPartInstance', showStyle1
  Rd2 = showstyle2
  Rd3 = showstyle1
  Rd4 = 'newInstance', showstyle1
To me it reads like this will continue infinites across Rd2, even though it is a different style.


I think there are two things to do here.
1) Make `rundownsBeforeThisInPlaylistSet` be an array. Looking back up the call trees, nothing seems to want it as a set, and all usages of the set convert it back to an array.
2) This logic needs some rethinking to check if it can be continued. Which should ideally be done outside of the nested for loops.

Reply:
I've made a change to look at all rundowns between the current part and the target part to check that they all match the same showstyle, as well as changing the set to an array.